internal/ethapi/api.go

```diff
@@ -879,6 +880,13 @@ func (s *PublicBlockChainAPI) Call(ctx context.Context, args CallArgs, blockNrOr
 	if err != nil {
 		return nil, err
 	}
+	if result.Err != nil {
```

Comment:
Perhaps it's not the best place to put these logic. Since jsonRPC and graphQL all use `DoCall` as the underlying API, maybe it's better to put the logic there.

Besides, we have the mirror APIs in `simulationBackend`. Please add them to it too.

Reply:
I moved it, however now EstimateGas contains the check/unpacking twice

Reply:
Which means the second unpacking is never hit (which is fine I think) but the error returned is now a normal error, not an estimateGasError{}